src/containerapp/azext_containerapp/dotnet_component_decorator.py

```diff
@@ -82,17 +82,25 @@ def delete(self):
 
     def _get_dotnet_component_if_exists(self, dotnet_component_name, environment_name, resource_group_name):
         try:
-            return self.client.show(self.cmd, dotnet_component_name, environment_name, resource_group_name)
+            return self.client.show(
+                cmd=self.cmd, resource_group_name=resource_group_name,
+                environment_name=environment_name, name=dotnet_component_name)
         except Exception as e:
             handle_non_404_status_code_exception(e)
             return None
 
-    def validate_arguments(self, dotnet_component_name, environment_name, resource_group_name, validation_error):
+    def validate_create_arguments(self, dotnet_component_name, environment_name, resource_group_name, validation_error):
         # Check if DotNet component already exists in environment
         existing_dotnet_component = self._get_dotnet_component_if_exists(dotnet_component_name, environment_name, resource_group_name)
         if existing_dotnet_component:
             raise ValidationError(validation_error)
 
+    def validate_delete_arguments(self, dotnet_component_name, environment_name, resource_group_name, validation_error):
+        # Check if DotNet component exists in environment
+        existing_dotnet_component = self._get_dotnet_component_if_exists(dotnet_component_name, environment_name, resource_group_name)
+        if not existing_dotnet_component:
+            raise ValidationError(validation_error)
+
```

Comment:
Recommend removing them both, sync offline, as this has been handle by CP.

Reply:
Removed. Thanks